src/main/scala/org/constellation/storage/CheckpointService.scala

```diff
@@ -271,17 +286,14 @@ class CheckpointService[F[_]: Sync: LiftIO](
       }
     }
 
-  private def syncPending(storage: StorageService[IO, CheckpointBlock],
-                          cb: CheckpointBlock)(implicit dao: DAO): F[Unit] =
-    Sync[F].delay {
-      storage.synchronized {
-        if (storage.contains(cb.baseHash).unsafeRunSync()) {
-          throw PendingAcceptance(cb.baseHash)
-        } else {
-          storage.put(cb.baseHash, cb).unsafeRunSync()
-        }
-      }
-    }
+  private def syncPending(storage: Ref[F, Set[String]], baseHash: String)(implicit dao: DAO): F[Unit] =
+    for {
+      pending <- storage.get
+      _ <- if (pending.contains(baseHash))
```

Comment:
as I've checked, it can be shorter using `IfMOps`:
```
storage.get.flatMap(_.contains(baseHash)).ifM {
 PendingAcceptance(baseHash).raiseError[F, Unit],
 storage.update(_ + baseHash)
}
```
```